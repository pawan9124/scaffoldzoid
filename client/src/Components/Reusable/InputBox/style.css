.reigster__name_label {
  padding: 5px;
  font-size: 14px;
  font-weight: 600;
  margin-bottom: 5px;
}
.register__name_input {
  height: 30px;
  width: 100%;
  outline-color: orange;
}
