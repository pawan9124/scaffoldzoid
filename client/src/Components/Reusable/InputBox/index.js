import React from "react";
import PropTypes from "prop-types";
import "./style.css";

function InputBox(props) {
  return (
    <>
      <label
        htmlFor=""
        className="reigster__name_label"
        data-test="inputBoxComponent"
      >
        {props.label}
      </label>
      <input
        type={props.type}
        className={`form-control register__name_input ${props.inputClassName} `}
        data-test="InputBoxComponent"
        id={props.id}
        placeholder={props.placeholder}
        onChange={(e) => props.handleChange(e.target.value)}
        name={props.name}
        value={props.value}
        disabled={props.disabledMode}
      />
      {props.errors && (
        <small className="register__validationErrors">{props.errors}</small>
      )}
    </>
  );
}

InputBox.propTypes = {
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default InputBox;
