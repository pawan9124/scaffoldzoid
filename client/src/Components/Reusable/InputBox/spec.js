import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../Utils";
import InputBox from "./index";
import renderer from "react-test-renderer";

let mockFunc = jest.fn();
const props = {
  type: "email",
  placeholder: "Please enter your email",
  handleChange: mockFunc,
  name: "Email"
};

describe("InputBox Component", () => {
  describe("Taking SnapShot", () => {
    it("Taking the snapshot of the InputBox Component", () => {
      const tree = renderer.create(<InputBox {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Checking PropTypes", () => {
    it("Should not throw a warning", () => {
      const propsError = checkProps(InputBox, props);
      expect(propsError).toBeUndefined();
    });
  });

  describe("Checking the Rendering", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<InputBox {...props} />);
    });

    it("Should render an InputBox", () => {
      const inputBox = findByTestAttr(wrapper, "InputBoxComponent");
      expect(inputBox.length).toBe(1);
    });

    it("Should emit callback on inputChange", () => {
      const inputBox = findByTestAttr(wrapper, "InputBoxComponent");
      const event = { target: { name: "Email", value: "donald@onedot.com" } };
      inputBox.simulate("change", event);
      const callback = mockFunc.mock.calls.length;
      expect(callback).toBe(1);
    });
  });
});
