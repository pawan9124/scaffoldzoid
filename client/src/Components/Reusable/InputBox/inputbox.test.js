import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../testUtils";
import InputBox from "./index";
import renderer from "react-test-renderer";

const props = {
  errors: undefined,
  handleChange: jest.fn(),
  label: "Retype Password",
  name: "password",
  placeholder: "Please retype password",
  type: "password",
  value: "",
};

describe("Taking snap shot", () => {
  it("Taking the snapshot of the Component", () => {
    const tree = renderer.create(<InputBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking the render of component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<InputBox {...props} />);
  });
  it("Should Render a component", () => {
    const button = findByTestAttr(wrapper, "inputBoxComponent");
    expect(button.length).toBe(1);
  });
});

describe("Checking PropTypes", () => {
  it("Should Not throw a warning", () => {
    const expectedProps = props;

    const propsError = checkProps(InputBox, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
