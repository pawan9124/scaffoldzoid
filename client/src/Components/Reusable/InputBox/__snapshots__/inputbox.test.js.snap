// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking snap shot Taking the snapshot of the Component 1`] = `
Array [
  <label
    className="reigster__name_label"
    data-test="inputBoxComponent"
    htmlFor=""
  >
    Retype Password
  </label>,
  <input
    className="form-control register__name_input undefined "
    data-test="InputBoxComponent"
    name="password"
    onChange={[Function]}
    placeholder="Please retype password"
    type="password"
    value=""
  />,
]
`;
